runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -914,10 +914,27 @@ chpl_taskID_t chpl_task_getId(void)
 void chpl_task_sleep(double secs)
 {
     if (qthread_shep() == NO_SHEPHERD) {
-        struct timespec delay;
-        delay.tv_sec = (time_t)(secs);
-        delay.tv_nsec = (long)(1e9*(secs - floor(secs)));
-        nanosleep(&delay, NULL);
+        struct timeval deadline;
+        struct timeval now;
+
+        //
+        // Figure out when this task can proceed again, and until then, keep
+        // yielding.
+        //
+        gettimeofday(&deadline, NULL);
+        deadline.tv_usec += (suseconds_t) ((secs - trunc(secs)) * 1.0e6);
+        if (deadline.tv_usec > 1000000) {
+            deadline.tv_sec++;
+            deadline.tv_usec -= 1000000;
+        }
+        deadline.tv_sec += (time_t) trunc(secs);
+
+        do {
+            chpl_task_yield();
```

Comment:
Right, there could someday be something more complicated than a sched_yield() in that case, though I don't have a prediction right now what it might be ...